core/src/main/scala/derevo/Derevo.scala

```diff
@@ -207,14 +208,36 @@ class Derevo(val c: blackbox.Context) {
 
     val callWithT = if (mode.passArgs) q"$call[$outTyp]" else call
 
+    def substNothing = {
+      def fixTpes(tpes: Seq[Tree]) = tpes.map {
+        case t if t.tpe == c.typeOf[Nothing] => tq"$outTyp"
```

Comment:
Well, this looks like very ad-hoc for the case where all but one type args are inferrable, and there could not be `Nothing` as a normal type param.
What do you think of convention that target type is always first in the list of the type parameters

Reply:
This can also fix the issue when someone really needs `Nothing` somewhere